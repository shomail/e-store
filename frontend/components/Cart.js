import React from 'react';
import { Query, Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import User from './User';
import CartStyles from './styles/CartStyles';
import Supreme from './styles/Supreme';
import CloseButton from './styles/CloseButton';
import SickButton from './styles/SickButton';

const LOCAL_STATE_QUERY = gql`
  query {
    cartOpen @client
  }
`;

const TOGGLE_CART_MUTATION = gql`
  mutation {
    toggleCart @client
  }
`;

const Cart = props => (
  <User>
    {({ data: { me } }) => {
      if (!me) return null;
      return (
        <Mutation mutation={TOGGLE_CART_MUTATION}>
          {toggleCart => (
            <Query query={LOCAL_STATE_QUERY}>
              {({ data }) => (
                <CartStyles open={data.cartOpen}>
                  <header>
                    <CloseButton onClick={toggleCart} title="close">
                      &times;
                    </CloseButton>
                    <Supreme>{me.name}'s Cart</Supreme>
                    <p>
                      You Have {me.cart.length} Item
                      {me.cart.length === 1 ? '' : 's'} in your Cart
                    </p>
                  </header>
                  <ul>
                    {me.cart.map(cartItem => (
                      <li>{cartItem.id}</li>
                    ))}
                  </ul>
                  <footer>
                    <p>$10.10</p>
                    <SickButton>Checkout</SickButton>
                  </footer>
                </CartStyles>
              )}
            </Query>
          )}
        </Mutation>
      );
    }}
  </User>
);

export default Cart;
export { LOCAL_STATE_QUERY, TOGGLE_CART_MUTATION };
