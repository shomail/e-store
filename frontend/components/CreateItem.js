import React, {Component} from 'react';
import {Mutation} from 'react-apollo'
import gql from 'graphql-tag'
import Form from './styles/Form'
import formatMoney from '../lib/formatMoney'

class CreateItem extends Component {
  state = {
    title: '',
    description: '',
    image: '',
    largeImage: '',
    price: 0
  }

  handleChange = (e) => {
    const { name, type, value } = e.target;
    const val = type === 'number' ? parseFloat(value) : value
    this.setState({ [name]: val })
  }

  render() {
    return (
      <Form>
        <fieldset>
          <label htmlFor="title">
            Title
            <input 
              type="text" 
              id="title" 
              name="title" 
              placeholder="Title" 
              required 
              value={this.state.title} 
              onChange={this.handleChange} 
            />
          </label>

          <label htmlFor="price">
            Price
            <input 
              type="number" 
              id="price" 
              name="price" 
              placeholder="Price" 
              required 
              value={this.state.price} 
              onChange={this.handleChange} 
            />
          </label>

          <label htmlFor="description">
            Description
            <textarea 
              id="description" 
              name="description" 
              placeholder="Enter desctiption" 
              required 
              value={this.state.description} 
              onChange={this.handleChange} 
            />
          </label>
          <button type="submit">Save</button>
        </fieldset>
      </Form>
    )
  }
}

export default CreateItem