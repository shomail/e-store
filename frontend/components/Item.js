import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Item extends Component {
  static PropTypes = {
    item: PropTypes.object.isRequired,
  };

  render() {
    return <div />;
  }
}
