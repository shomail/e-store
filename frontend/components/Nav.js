import Link from 'next/link';

const Nav = () => (
  <div>
    <Link href="/">
      <a>Home</a>
    </Link>
    <Link href="/sell">
      <a>Sell you Items</a>
    </Link>
  </div>
);

export default Nav;
