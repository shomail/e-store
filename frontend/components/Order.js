import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import { format } from 'date-fns';
import Head from 'next/head';
import formatMoney from '../lib/formatMoney';
import Error from './ErrorMessage';
import OrderStyles from './styles/OrderStyles';

const SINGLE_ORDER_QUERY = gql`
  query SINGLE_ORDER_QUERY($id: ID!) {
    order(id: $id) {
      id
      charge
      total
      createdAt
      user {
        id
      }
      items {
        id
        title
        description
        price
        image
        quantity
      }
    }
  }
`;

class Order extends Component {
  static propTypes = {
    id: PropTypes.string.isRequired,
  };

  render() {
    const { id } = this.props;
    return (
      <Query query={SINGLE_ORDER_QUERY} variables={{ id }}>
        {({ data, error, loading }) => {
          if (error) return <Error error={error} />;
          if (loading) return <p>Loading...</p>;
          console.log(data);
          return (
            <div>
              <p>Order Id: {id}</p>
            </div>
          );
        }}
      </Query>
    );
  }
}

export default Order;
