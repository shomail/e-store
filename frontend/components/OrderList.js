import React, { Component } from 'react';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import { formatDistance } from 'date-fns';
import Link from 'next/link';
import styled from 'styled-components';
import formatMoney from '../lib/formatMoney';
import OrderItemStyles from './styles/OrderItemStyles';

// const QUERY_ORDER_LIST = {
//     query orderList {

//     }
// }

class OrderList extends Component {
  render() {
    return <p>Orders List</p>;
  }
}

export default OrderList;
