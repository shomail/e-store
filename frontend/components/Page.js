import React, {Component} from 'react';

export default class Page extends Component {
  render() {
    return (
      <div>
        <nav>Navigation</nav>
        {this.props.children}
      </div>
    )
  }
}