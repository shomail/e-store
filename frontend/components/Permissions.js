import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import Error from './ErrorMessage';
import Table from './styles/Table';

const possiblePermissions = [
  'ADMIN',
  'USER',
  'ITEMCREATE',
  'ITEMUPDATE',
  'ITEMDELETE',
  'PERMISSIONUPDATE',
];

const ALL_USERS_QUERY = gql`
  query ALL_USERS_QUERY {
    users {
      id
      name
      email
      permissions
    }
  }
`;

const Permissions = props => (
  <Query query={ALL_USERS_QUERY}>
    {({ data, loading, error }) => (
      <>
        <Error error={error} />
        <h2>Manage Permissions</h2>
        <Table>
          <thead>
            <tr>
              <th>Name</th>
              <th>Email</th>
              {possiblePermissions.map(permission => (
                <th>{permission}</th>
              ))}
              <th>&nbsp;</th>
            </tr>
          </thead>
          <tbody>
            {data.users.map(user => (
              <User user={user} />
            ))}
          </tbody>
        </Table>
      </>
    )}
  </Query>
);

class User extends React.Component {
  render() {
    const {user} = this.props;
    return (
      <tr>
        <td>{user.name}</td>
        <td>{user.email}</td>
      </tr>
    );
  }
}

export default Permissions;
