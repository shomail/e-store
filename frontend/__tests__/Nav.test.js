import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import { MockedProvider } from 'react-apollo/test-utils';
import { CURRENT_USER_QUERY } from '../components/User';
import Nav from '../components/Nav';
import { fakeUser } from '../lib/testUtils';

const notSignedInMocks = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: { data: { me: null } },
  },
];

const signedInMocks = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: { data: { me: fakeUser() } },
  },
];

describe('<Nav />', () => {
  it('renders guest user nav', async () => {
    const wrapper = mount(
      <MockedProvider mocks={notSignedInMocks}>
        <Nav />
      </MockedProvider>
    );
    await wait();
    wrapper.update();
    const nav = wrapper.find('[data-test="nav"]');
    // console.log(nav.debug());
    expect(toJSON(nav)).toMatchSnapshot();
  });
  it('renders signed in user nav', async () => {
    const wrapper = mount(
      <MockedProvider mocks={signedInMocks}>
        <Nav />
      </MockedProvider>
    );
    await wait();
    wrapper.update();
    const nav = wrapper.find('[data-test="nav"]');
    expect(toJSON(nav)).toMatchSnapshot();
  });
});
