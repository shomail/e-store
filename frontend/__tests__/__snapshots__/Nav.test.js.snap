// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> renders guest user nav 1`] = `
Array [
  <NavStyles
    data-test="nav"
  >
    <ul
      className="NavStyles-sc-11c0d2g-0 dDiqeM"
      data-test="nav"
    >
      <Link
        href="/items"
      >
        <a
          href="/items"
          onClick={[Function]}
        >
          Shop
        </a>
      </Link>
      <Link
        href="/signup"
      >
        <a
          href="/signup"
          onClick={[Function]}
        >
          Sign In
        </a>
      </Link>
    </ul>
  </NavStyles>,
  <ul
    className="NavStyles-sc-11c0d2g-0 dDiqeM"
    data-test="nav"
  >
    <Link
      href="/items"
    >
      <a
        href="/items"
        onClick={[Function]}
      >
        Shop
      </a>
    </Link>
    <Link
      href="/signup"
    >
      <a
        href="/signup"
        onClick={[Function]}
      >
        Sign In
      </a>
    </Link>
  </ul>,
]
`;
